Add Development environment info to error page

rowanmiller opened the pull request:
If you are not running from VS, then by default you are not in
development environment (until you set an environment variable). Adding
some info to the non-Development error page to help out folks developing
outside of VS.

Resolves #351 

cc @coolcsh for his input on whether the text would have helped him, since he ran into this prepping for a demo

phenning commented:
Any feed back on this? @rustd, @sayedihashimi

rustd commented:
I had a chat with @rowanmiller The simplest change is to change the Error page to show a message always saying that switching to development environment will give you more details.

Here is what the page would look like?

``` html
    <p>
        Swapping to <strong>Development</strong> environment will display more detailed information on the error that occurred.
    </p>
    <p>
        Development environment can be enabled by setting the <strong>Hosting:Environment</strong> environment variable to <strong>Development</strong>, and restarting the application.
    </p>
```

rustd commented:
@rowanmiller are you going to update this PR or should we go ahead with my proposed change?

rowanmiller commented:
I can update it quickly with your suggestion... gimme a couple of min :smile:

rowanmiller commented:
Updated, ready for review

rustd commented on src/BaseTemplates/StarterWeb/Views/Shared/Error.cshtml:
```diff
@@ -4,3 +4,11 @@
+    Swapping to <strong>Development</strong> environment will display more detailed information about the error that occurred.
+</p>
+<p>
+    Development environment can be enabled by setting the <strong>Hosting:Environment</strong> environment variable to <strong>Development</strong>, and restarting the application.
```
This name should match whatever is the latest in RC2. Looks good otherwise

rowanmiller replied:
Do you know what the name is for RC2? I've been digging thru hosting code and it's a bit of a rabbit maze to work out exactly what env variable controls the name :smile:

rustd replied:
ASPNETCORE_ENVIRONMENT from https://github.com/aspnet/Hosting/pull/640

rowanmiller replied:
Thanks! Am I good to push this one with that change?

rustd replied:
:shipit: We will update it later if there is a different name

blowdart commented:
At least point out that this should never be done on a live server due to security concerns.

rowanmiller commented:
@blowdart do you have some text we have used elsewhere? Or something you want us to use?

blowdart commented:
Well I'd rather you didn't add this at all. It's the same problem we had in ASP.NET 4. People would follow the advice blindly. It encourages poor practices.

rustd commented:
Hmm. Let's hold off on this for RC2 and see if we have a better solution in RTM.

rowanmiller committed:
Add Development environment info to error page

If you are not running from VS, then by default you are not in
development environment (until you set an environment variable). Adding
some info to the non-Development error page to help out folks developing
outside of VS.

Resolves #351

rowanmiller commented:
Updated with some text about why not to do this... and the text is before the bit that tells you how to do it. I think we should push to get this into RC2 since it is a confusing issue that folks have hit and there is no way to discover this if you run outside of VS (other than reading docs).

rustd commented:
@blowdart ping. Is Rowan's proposal good enough for RC2 or is there a better solution here.

blowdart commented:
&lt;strong&gt; the should not and I'll acquiesce.

rowanmiller commented:
Done. @rustd will someone on your side merge?

rustd commented:
@phenning can merge.
